project/generate-dockerfile.sh

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+set -e
+
+cat <<-'EOF'
+#
+# DO NOT EDIT - GENERATED BY project/generate-dockerfile.sh
+#
+# This file describes the standard way to build Docker, using docker
+#
+# Usage:
+#
+# # Assemble the full dev environment. This is slow the first time.
+# docker build -t docker .
+#
+# # Mount your source in an interactive container for quick testing:
+# docker run -v `pwd`:/go/src/github.com/docker/docker --privileged -i -t docker bash
+#
+# # Run the test suite:
+# docker run --privileged docker hack/make.sh test
+#
+# # Publish a release:
+# docker run --privileged \
+#  -e AWS_S3_BUCKET=baz \
+#  -e AWS_ACCESS_KEY=foo \
+#  -e AWS_SECRET_KEY=bar \
+#  -e GPG_PASSPHRASE=gloubiboulga \
+#  docker hack/release.sh
+#
+# Note: Apparmor used to mess with privileged mode, but this is no longer
+# the case. Therefore, you don't have to disable it anymore.
+#
+
+EOF
+
+if [ "$ARCH" == "ppc64le" ]; then
+	cat <<-'EOF'
+		FROM gccgo_image_14
+	EOF
+elif [ "$ARCH" == "s390x" ]; then
+	cat <<-'EOF'
+		FROM gccgo-image
+		RUN echo "deb http://ftp.de.debian.org/debian wheezy-backports main " >> /etc/apt/sources.list
+	EOF
+elif [ "$ARCH" == "amd64" ] && [ "$GCCGO" == "true" ]; then
+	cat <<-'EOF'
+		FROM gccgo_image_14
```

Comment:
Until we have a chance to have true multi-arch support in a future registry, I think we should use tagging and some name conformity to make it cleaner: `gccgo14:ppc64le` and `gccgo14:s390x` for example?